WooCommerce/Classes/Authentication/Epilogue/StorePickerViewModel.swift

```diff
@@ -0,0 +1,207 @@
+        state = StorePickerState(sites: resultsController.fetchedObjects)
+    }
+
+    func synchronizeSites(selectedSiteID: Int64?, onCompletion: @escaping (Result<Void, Error>) -> Void) {
+        let syncStartTime = Date()
+        let isJetpackConnectionPackageSupported = ServiceLocator.featureFlagService.isFeatureFlagEnabled(.jetpackConnectionPackageSupport)
+        let action = AccountAction
+            .synchronizeSites(selectedSiteID: selectedSiteID,
+                              isJetpackConnectionPackageSupported: isJetpackConnectionPackageSupported) { result in
+                switch result {
+                case .success(let containsJCPSites):
+                    if containsJCPSites {
+                        let syncDuration = round(Date().timeIntervalSince(syncStartTime) * 1000)
+                        ServiceLocator.analytics.track(.jetpackCPSitesFetched, withProperties: ["duration": syncDuration])
+                    }
+                    onCompletion(.success(()))
+                case .failure(let error):
+                    onCompletion(.failure(error))
+                }
+            }
+        stores.dispatch(action)
+    }
+}
+
+// MARK: - Table view configs
+extension StorePickerViewModel {
+    /// Indicates if there is more than one Store.
+    ///
+    var multipleStoresAvailable: Bool {
+        return resultsController.fetchedObjects.count > 1
+    }
+
+    // Results Table's Separator Style
+    ///
+    var separatorStyle: UITableViewCell.SeparatorStyle {
+        guard resultsController.numberOfObjects > 0 else {
+            return .none
+        }
+        return .singleLine
+    }
+
+    /// Number of sections in the table view
+    ///
+    var numberOfSections: Int {
+        guard resultsController.numberOfObjects > 0 else {
+            return 1
+        }
+        return resultsController.sections.count
+    }
+
+    /// Title of the section at the specified index
+    ///
+    func titleForSection(at index: Int) -> String? {
+        guard resultsController.numberOfObjects > 0 else {
+            return nil
+        }
+
+        guard let rawStatus = resultsController.sections[safe: index]?.name else {
+            return nil
+        }
+
+        let isWooCommerceActive = (rawStatus as NSString).boolValue
+        if isWooCommerceActive {
+            return multipleStoresAvailable ? Localization.pickStore : Localization.connectedStore
+        } else {
+            return Localization.otherSites
+        }
+    }
+
+    /// Number of rows in a specified table view section index.
+    ///
+    func numberOfRows(inSection sectionIndex: Int) -> Int {
+        guard resultsController.numberOfObjects > 0 else {
+            return 1
+        }
+        return resultsController.sections[safe: sectionIndex]?.objects.count ?? 0
+    }
+
+    /// Returns the site to be displayed at a given IndexPath
+    ///
+    func site(at indexPath: IndexPath) -> Yosemite.Site? {
```

Comment:
nit: is `Yosemite` needed?

```suggestion
    func site(at indexPath: IndexPath) -> Site? {
```

Reply:
Removed in 38e1d1dcb28d2b6c8c6d4932f05f6b8798b79451.